kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/config/ConfigParserTest.java

```diff
@@ -268,6 +271,67 @@ void shouldDetectDuplicateClusterNodeNames() {
 
     }
 
+    @Test
+    void systemPropertyExpansion() {
+
+        System.setProperty("bootstrap", "zzzz");
+        System.setProperty("port", "1234");
+        // Given
+        Configuration configuration = configParser.parseConfiguration("""
+                virtualClusters:
+                  demo1:
+                    targetCluster:
+                      bootstrap_servers: ${sys:bootstrap}:${sys:port}
+                    clusterNetworkAddressConfigProvider:
+                      type: PortPerBrokerClusterNetworkAddressConfigProvider
+                      config:
+                        bootstrapAddress: cluster:9193
+                """);
+
+        var actualValidClusters = configuration.virtualClusters();
+
+        // Then
+        assertThat(actualValidClusters)
+                .flatExtracting("demo1")
+                .singleElement(InstanceOfAssertFactories.type(VirtualCluster.class))
+                .extracting(VirtualCluster::targetCluster)
+                .extracting(TargetCluster::bootstrapServers)
+                .isEqualTo("zzzz:1234");
+
+        System.clearProperty("bootstrap");
```

Comment:
n.b this won't be run if the test fails

Reply:
Yes, it wasn't supposed to be commit quality code.

Reply:
I was overly terse, I was just trying to mark for all readers where the Junit pioneer extension would make a difference.